vmware-tanzu/community-edition
# harbor package 2.3.3 support secretName for tlsCertificate

Signed-off-by: Shengwen Yu <yshengwen@vmware.com>

## What this PR does / why we need it

This PR is to introduce a new configuration called "secretName" under "tlsCertificate" so that Harbor package 2.3.3 
can auto reload ca.crt, tls.crt, tls.key info.

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
Introducing a new configuration called "secretName" under "tlsCertificate" 
in order to auto load ca.crt, tls.crt, tls.key info for Harbor 2.3.3
```

## Which issue(s) this PR fixes

Fixes: #2553

## Describe testing done for PR

unit test:
`cd addons/packages/harbor/2.3.3/test`
`make test`

functional test:
* create a secret call "harbor-tls-2" with ca.crt, tls.crt, tls.key and duration (valid for) 1h through openssl tool
* set "tlsCertificate.secretName" in harbor-data-values.yaml file to be "harbor-tls-2"
* install Harbor package 2.3.3 with harbor-data-values.yaml
* login into Harbor portal, check the certificate validation, it's only valid within 1h after it's created
* using openssl to generate certificate info valid through 2 hours and update secret "harbor-tls" 
* login to Harbor portal, check the certificate validation, it's should be valid within 2 hours after it's created

## Special notes for your reviewer
